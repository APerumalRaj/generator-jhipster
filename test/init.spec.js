const path = require('path');
const expect = require('expect');
const normalizePath = require('normalize-path');

const { skipPrettierHelpers: helpers } = require('./utils/utils');

describe('JHipster init generator', () => {
  describe('with default options', () => {
    let runResult;
    before(async () => {
      runResult = await helpers.run(path.join(__dirname, '../generators/init'));
    });
    it('should create expected files', () => {
      expect(runResult.getStateSnapshot()).toMatchSnapshot();
    });
  });
  describe('with custom prompt values', () => {
    let runResult;
    const promptValues = {
      projectName: 'Beautiful Project',
      baseName: 'BeautifulProject',
      prettierDefaultIndent: 4,
      prettierJavaIndent: 2,
    };
    describe('and default options', () => {
      before(async () => {
        runResult = await helpers.run(path.join(__dirname, '../generators/init')).withPrompts(promptValues);
      });
      it('should write custom config to .yo-rc.json', () => {
        const yoFile = normalizePath(path.join(runResult.cwd, '.yo-rc.json'));
        expect(runResult.getSnapshot(file => file.path === yoFile)).toMatchSnapshot();
      });
    });
    describe('and skipPrompts option', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .run(path.join(__dirname, '../generators/init'))
          .withOptions({ skipPrompts: true })
          .withPrompts(promptValues);
      });
      it('should write default values to .yo-rc.json', () => {
        const yoFile = normalizePath(path.join(runResult.cwd, '.yo-rc.json'));
        expect(runResult.getSnapshot(file => file.path === yoFile)).toMatchSnapshot();
      });
    });
    describe('and existing config', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .run(path.join(__dirname, '../generators/init'))
          .withOptions({ localConfig: { baseName: 'existing' } })
          .withPrompts(promptValues);
      });
      it('should not write custom prompt values', () => {
        const yoFile = normalizePath(path.join(runResult.cwd, '.yo-rc.json'));
        expect(runResult.getSnapshot(file => file.path === yoFile)).toMatchSnapshot();
      });
    });
    describe('and askAnsweredOption on a exiting project', () => {
      let runResult;
      before(async () => {
        runResult = await helpers
          .run(path.join(__dirname, '../generators/init'))
          .withOptions({ askAnswered: true, localConfig: { baseName: 'existing' } })
          .withPrompts(promptValues);
      });
      it('should write custom prompt values', () => {
        const yoFile = normalizePath(path.join(runResult.cwd, '.yo-rc.json'));
        expect(runResult.getSnapshot(file => file.path === yoFile)).toMatchSnapshot();
      });
    });
  });
  describe('with custom', () => {
    describe('skipCommitHook option', () => {
      let runResult;
      before(async () => {
        runResult = await helpers.run(path.join(__dirname, '../generators/init')).withOptions({ skipCommitHook: true });
      });
      it('should create expected files', () => {
        expect(runResult.getStateSnapshot()).toMatchSnapshot();
      });
    });
  });
});
