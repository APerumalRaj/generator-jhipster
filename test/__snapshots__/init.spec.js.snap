// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JHipster init generator with custom prompt values and askAnsweredOption on a exiting project should write custom prompt values 1`] = `
Object {
  ".yo-rc.json": Object {
    "contents": "{
  \\"generator-jhipster\\": {
    \\"baseName\\": \\"BeautifulProject\\",
    \\"jhipsterVersion\\": \\"0.0.0\\",
    \\"projectName\\": \\"Beautiful Project\\",
    \\"prettierDefaultIndent\\": 4,
    \\"prettierJavaIndent\\": 2
  }
}
",
    "stateCleared": "modified",
  },
}
`;

exports[`JHipster init generator with custom prompt values and default options should write custom config to .yo-rc.json 1`] = `
Object {
  ".yo-rc.json": Object {
    "contents": "{
  \\"generator-jhipster\\": {
    \\"jhipsterVersion\\": \\"0.0.0\\",
    \\"projectName\\": \\"Beautiful Project\\",
    \\"baseName\\": \\"BeautifulProject\\",
    \\"prettierDefaultIndent\\": 4,
    \\"prettierJavaIndent\\": 2
  }
}
",
    "stateCleared": "modified",
  },
}
`;

exports[`JHipster init generator with custom prompt values and existing config should not write custom prompt values 1`] = `
Object {
  ".yo-rc.json": Object {
    "contents": "{
  \\"generator-jhipster\\": {
    \\"baseName\\": \\"existing\\",
    \\"jhipsterVersion\\": \\"0.0.0\\",
    \\"prettierDefaultIndent\\": 2,
    \\"prettierJavaIndent\\": 4,
    \\"projectName\\": \\"JHipster project\\"
  }
}
",
    "stateCleared": "modified",
  },
}
`;

exports[`JHipster init generator with custom prompt values and skipPrompts option should write default values to .yo-rc.json 1`] = `
Object {
  ".yo-rc.json": Object {
    "contents": "{
  \\"generator-jhipster\\": {
    \\"jhipsterVersion\\": \\"0.0.0\\",
    \\"prettierDefaultIndent\\": 2,
    \\"prettierJavaIndent\\": 4,
    \\"projectName\\": \\"JHipster project\\"
  }
}
",
    "stateCleared": "modified",
  },
}
`;

exports[`JHipster init generator with default options should create expected files 1`] = `
Object {
  ".editorconfig": Object {
    "stateCleared": "modified",
  },
  ".gitattributes": Object {
    "stateCleared": "modified",
  },
  ".gitignore": Object {
    "stateCleared": "modified",
  },
  ".huskyrc": Object {
    "stateCleared": "modified",
  },
  ".lintstagedrc.js": Object {
    "stateCleared": "modified",
  },
  ".prettierignore": Object {
    "stateCleared": "modified",
  },
  ".prettierrc": Object {
    "stateCleared": "modified",
  },
  ".yo-rc.json": Object {
    "stateCleared": "modified",
  },
  "README.md": Object {
    "stateCleared": "modified",
  },
  "package.json": Object {
    "stateCleared": "modified",
  },
}
`;
